addons/mail/static/src/core/web/chatter.js

```diff
@@ -142,37 +142,29 @@ export class Chatter extends Component {
         );
 
         onMounted(async () => {
-            if (this.props.threadId) {
-                this.state.thread = this.store.Thread.insert({
-                    id: this.props.threadId,
-                    model: this.props.threadModel,
-                    name: this.props.webRecord?.data?.display_name || undefined,
-                });
-            }
-            await this.load(this.props.threadId, [
+            await this.load(this.props.threadModel, this.props.threadId, [
```

Comment:
I don't think we should `await` for load to `changeThread`, we can display what we already have immediately rather than keep displaying the old thread.

Reply:
but what if the documents attached were removed or something changed while having a bad connection. This is lead to some problems unwanted.

Reply:
But then it should also be done in update props??
Having different code in mounted and update props makes no sense, it should be exactly the same.

Reply:
I prefer to add await in update props so that it is safer

Reply:
But that means we can do "next" on a document, and we will be on a new document, and still see the data of the previous thread. I think that can be very confusing.

Reply:
At the very minimum thread should be set to undefined then before the load to make sure we don't display previous data.

Reply:
it is more or less like a `loading status`. I can do like that. Thanks.

Reply:
But it will flicker when changing documents and we don't want that either. The top level condition `t-if="state.thread"` needs to be adapted to avoid flickers and protect parts of the template more specifically and better handle thread loading.

Reply:
But I still think this is not a good idea. We should just immediately show the data we already have for a thread.
99% of the time it won't change, and update will come fast as well.